maestro-ios-xctest-runner/maestro-driver-iosUITests/Routes/Handlers/ViewHierarchyHandler.swift

```diff
@@ -58,30 +58,23 @@ struct ViewHierarchyHandler: HTTPHandler {
             springboardHierarchy = nil
         }
 
-        let appHierarchy = try appHierarchy(XCUIApplication(bundleIdentifier: appId))
+        let appHierarchy = try getHierarchyWithFallback(app)
 
         return AXElement(children: [
             springboardHierarchy,
             appHierarchy,
         ].compactMap { $0 })
     }
     
-    func getRunningAppId(_ runningAppIds: [String]) -> String {
-        return runningAppIds.first { appId in
-            let app = XCUIApplication(bundleIdentifier: appId)
-            
-            return app.state == .runningForeground
-        } ?? ViewHierarchyHandler.springboardBundleId
+    func getRunningApp(_ runningAppIds: [String]) -> XCUIApplication? {
+        runningAppIds
+            .map { XCUIApplication(bundleIdentifier: $0) }
+            .first { app in app.state == .runningForeground }
```

Comment:
Why this function doesn't return the value?

Reply:
It does?